napari/utils/_injection.py

```diff
@@ -35,31 +40,75 @@ def get_accessor(type_: Type[T]) -> Optional[Callable[[], Optional[T]]]:
     `inject_napari_dependencies`, allows us to inject current napari objects
     into functions based on type hints.
     """
-    from ..components import LayerList
-    from ..layers import Layer
-    from ..viewer import Viewer, current_viewer
+    if type_ in _ACCESSORS:
+        return _ACCESSORS[type_]
 
     if isinstance(type_, type):
-        if issubclass(type_, Layer):
-            return _get_active_layer
-        if issubclass(type_, Viewer):
-            return current_viewer
-        if issubclass(type_, LayerList):
-            return _get_active_layer_list
+        for key, val in _ACCESSORS.items():
+            if issubclass(type, key):
```

Comment:
Do you mean the following?

```suggestion
            if issubclass(type_, key):
```

I.e. this function also returns an accessor when the given type is a subclass of a key. If so, might be good to cover such a case with a small test.


I don't see how `type` could be a subclass of some custom user defined type, but that could definitely be my Python ignorance speaking.

Reply:
good eye!! I found that also in a later PR based on this one